DFA fails for small example with loop when using .fum file

[test_no_targets.zip](https://github.com/tokiwa-software/fuzion/files/10717298/test_no_targets.zip)

The attached example, when using module file and C backend, causes an Error during DFA:

```
 > unzip test_no_targets.zip 
Archive:  test_no_targets.zip
   creating: test_no_targets/
   creating: test_no_targets/src/
 extracting: test_no_targets/src/a.fz  
  inflating: test_no_targets/Makefile  
 > cd test_no_targets
 > make
echo "m is _ := a" | ../build/bin/fz -sourceDirs=src -
echo "m is _ := a" | ../build/bin/fz -c -sourceDirs=src -
../build/bin/fz -saveLib=a.fum -sourceDirs=src
 + a.fum
echo "m is _ := a" | ../build/bin/fz -modules=a -moduleDirs=. -
echo "m is _ := a" | ../build/bin/fz -c -modules=a -moduleDirs=. -

error 1: NYI: in a no targets for Call to hasInterval i32.infix ...ref as_stream.has_next target boxed(i32.infix ...ref as_stream):i32.infix ...as_stream

one error.
make: *** [Makefile:16: mod_c] Fehler 1

```

--- a/tests/reg_issue1109/Makefile
+++ b/tests/reg_issue1109/Makefile
@@ -1,0 +1,28 @@
+FUZION_OPTIONS ?=
+FUZION = ../../bin/fz $(FUZION_OPTIONS)
+
+all: jvm c int
+
+jvm: mod_jvm
+	echo "m is _ := a" | $(FUZION) -jvm -sourceDirs=src -
+
+c: mod_c
+	echo "m is _ := a" | $(FUZION) -c -sourceDirs=src -
+
+int: mod_int
+	echo "m is _ := a" | $(FUZION) -interpreter -sourceDirs=src -
+
+a.fum:
+	$(FUZION) -saveLib=a.fum -sourceDirs=src
+
+mod_jvm: a.fum
+	echo "m is _ := a" | $(FUZION) -jvm -modules=a -moduleDirs=. -
+
+mod_c: a.fum
+	echo "m is _ := a" | $(FUZION) -c -modules=a -moduleDirs=. -
+
+mod_int: a.fum
+	echo "m is _ := a" | $(FUZION) -interpreter -modules=a -moduleDirs=. -
+
+clean:
+	rm a.fum m m.c *~ */*~

--- a/tests/reg_issue1109/src/a.fz
+++ b/tests/reg_issue1109/src/a.fz
@@ -1,0 +1,2 @@
+public a is
+  for i in 0.. while i<1 do

